index.html

```diff
@@ -140,76 +190,254 @@ <h2>
+          Promise&lt;void&gt; setAppBadge(optional [EnforceRange] unsigned long long contents);
+          Promise&lt;void&gt; clearAppBadge();
         };
+
+        Navigator includes NavigatorBadge;
+        WorkerNavigator includes NavigatorBadge;
       </pre>
       <p>
-        The <dfn>Badge</dfn> namespace provides functions for setting and
-        clearing the application badge indicator.
+        User agents that never display document badges SHOULD NOT expose the
+        {{Navigator/setClientBadge()}} and {{Navigator/clearClientBadge()}}
+        methods. Similarly, user agents that never display application badges
+        SHOULD NOT expose the {{NavigatorBadge/setAppBadge()}} and
+        {{NavigatorBadge/clearAppBadge()}} methods.
+      </p>
+      <p class="note">
+        This is important as a feature-detection mechanism to allow sites to
+        determine whether setting a badge will have any effect, based on the
+        presence of the methods. In particular, sites can use the absence of
+        {{Navigator/setClientBadge()}} as a condition for displaying a fallback
+        badge in the page's [^title^] or <a data-cite=
+        "html#rel-icon"><code>icon</code></a>.
       </p>
+      <div class="issue" data-number="60"></div>
+      <div class="issue" data-number="61"></div>
       <section>
         <h3>
-          <dfn>set()</dfn> function
+          <dfn data-dfn-for="Navigator">setClientBadge()</dfn> method
         </h3>
         <p>
-          When the <code>set()</code> function is called with argument
-          <var>contents</var>, run the following steps:
+          When the {{Navigator/setClientBadge()}} method is called with
+          argument |contents:unsigned long long|:
         </p>
         <ol>
-          <li>If <var>contents</var> is omitted, set the application's badge to
-          <a>flag</a>.
+          <li>Return [=a promise resolved with=] undefined, and [=in
+          parallel=]:
+            <ol>
+              <li>Let |document:Document| be the [=current settings object=]'s
+              [=environment settings object/responsible document=].
+              </li>
+              <li>If |contents| is omitted, set the badge associated with
+              |document| to <a>flag</a>.
+              </li>
+              <li>Else, if |contents| is 0, set the badge associated with
+              |document| to <a>nothing</a>.
+              </li>
+              <li>Else, set the badge associated with |document| to |contents|.
+              </li>
+            </ol>
           </li>
-          <li>If <var>contents</var> is 0, set the application's badge to
+        </ol>
+      </section>
+      <section>
+        <h3>
+          <dfn data-dfn-for="Navigator">clearClientBadge()</dfn> method
+        </h3>
+        <p>
+          When the {{Navigator/clearClientBadge()}} method is called, return
+          [=a promise resolved with=] undefined, and [=in parallel=], set the
```

Comment:
As above... we need enough scope in the spec that clearing could error... if it doesn't error, then we can resolve the promise with undefined.